src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -195,6 +196,40 @@ public virtual SecurityToken CreateToken(SecurityTokenDescriptor tokenDescriptor
             return new Saml2SecurityToken(assertion);
         }
 
+        /// <summary>
+        /// Validates a token.
+        /// ArgumentNullException (parameters are null or empty) and ArgumentException (token length exceeds the maximum size) will be thrown on invalid parameters.
+        /// On a validation failure, no exception will be thrown; instead, the exception will be set in the returned TokenValidationResult.Exception property.
+        /// Callers should always catch and handle invalid argument exceptions, and check the TokenValidationResult.IsValid property to verify the validity of the result.
+        /// </summary>
+        /// <param name="token">The token to be validated.</param>
+        /// <param name="validationParameters">A <see cref="TokenValidationParameters"/> required for validation.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="token"/> is null or empty.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="validationParameters"/> is null or empty.</exception>
+        /// <exception cref="ArgumentException">'token.Length' is greater than <see cref="TokenHandler.MaximumTokenSizeInBytes"/>.</exception>
+        /// <returns>A <see cref="TokenValidationResult"/></returns>
+        public override Task<TokenValidationResult> ValidateTokenAsync(string token, TokenValidationParameters validationParameters)
+        {
+            try
+            {
+                var claimsPrincipal = ValidateToken(token, validationParameters, out var validatedToken);
+                return Task.FromResult(new TokenValidationResult
+                {
+                    SecurityToken = validatedToken,
+                    ClaimsIdentity = claimsPrincipal?.Identity as ClaimsIdentity,
```

Comment:
We need to think about how to deal with ValidateToken returns a ClaimsPrincipal that has multiple identities.

Reply:
It is now always returning the first identity as discussed.